activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -1032,7 +1032,8 @@ def build_statement_pool
 
         def can_perform_case_insensitive_comparison_for?(column)
           @case_insensitive_cache ||= {}
-          @case_insensitive_cache[column.sql_type] ||= begin
+          @case_insensitive_cache[column.sql_type] =
+            @case_insensitive_cache.fetch(column.sql_type) do
```

Comment:
```suggestion
          @case_insensitive_cache.fetch(column.sql_type) do
            @case_insensitive_cache[column.sql_type] =
```

Reply:
Something fishy is going on [here](https://buildkite.com/rails/rails/builds/91213#0184ada5-d3c3-413b-851f-798616f6ea3e/1101-1112).

With this suggested change, we're basically getting rid of a redundant `c[t] = c[t]` when there's a `t` key in `c` already.

I'm uncertain if the same conditions as in "do not modify the array during iteration" apply here, but I'd rather keep on the safe side.

I'll revert this change if you don't object.

Reply:
No it should be fine. I can have a look at the CI failure in a bit.

Reply:
Ah, I think it's missing a `begin/end`